From: Daniel van Vugt <daniel.van.vugt@canonical.com>
Date: Tue, 2 Jan 2024 10:23:29 +0100
Subject: Set default device timeout to 8.0, otherwise graphics fails.

Original-Author: Dimitri John Ledkov <xnox@ubuntu.com>
Bug-Ubuntu: https://bugs.launchpad.net/bugs/1838725
Forwarded: not-needed
Last-Update: 2020-04-01

The patch is needed because Ubuntu doesn't install the upstream configuration
file defaults.plymouth (which contains DeviceTimeout=8), but uses
alternatives instead.
Last-Update: 2020-04-01
---
 src/main.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/main.c b/src/main.c
index e2f6f86..5c5c891 100644
--- a/src/main.c
+++ b/src/main.c
@@ -2472,6 +2472,10 @@ main (int    argc,
         find_system_default_splash (&state);
         find_distribution_default_splash (&state);
 
+        /* Device timeout may not be NAN or zero */
+        if (isnan (state.device_timeout) || state.device_timeout <= 0.0)
+                state.device_timeout = 8.0;
+
         if (ply_kernel_command_line_has_argument ("plymouth.ignore-serial-consoles") ||
             ignore_serial_consoles == true)
                 device_manager_flags |= PLY_DEVICE_MANAGER_FLAGS_IGNORE_SERIAL_CONSOLES;
